core/dao/abstract-dao.js

```diff
@@ -0,0 +1,69 @@
+var Montage = require("montage/core/core").Montage,
+    FreeNASService = require("core/service/freenas-service").FreeNASService;
+
+exports.AbstractDao = Montage.specialize({
+    _instance: {
+        value: null
+    },
+
+    init: {
+        value: function() {
+            throw new Error("Dao must override `init` method.");
+        }
+    },
+
+    list: {
+        value: function() {
+            if (this._model) {
```

Comment:
this checking could be done in a method and be called by `list`, `getEmptyList`, `getNewInstance` and `save`